Client/Application/AppDelegate.swift

```diff
@@ -185,6 +186,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             NSUserDefaults.standardUserDefaults().setObject(firefoxUA, forKey: "UserAgent")
         }
         FaviconFetcher.userAgent = firefoxUA!
+        Sync15StorageClient.userAgent = firefoxUA!
```

Comment:
Extract all of the code that computes `firefoxUA` into a method, somewhere in `Shared` (Utils).

Then you can just use it once in `setUserAgent`, and again in `StorageClient`.